turbo/snapshotsync/freezeblocks/block_snapshots.go

```diff
@@ -725,6 +724,43 @@ func (s *RoSnapshots) buildMissedIndicesIfNeed(ctx context.Context, logPrefix st
 	return nil
 }
 
+func (s *RoSnapshots) Delete(fileName string) error {
+	if s == nil {
+		return nil
+	}
+	s.lockSegments()
```

Comment:
- let's use `.View()` instead of manual locking
- let's view.Close() in defer somehow